### ISSUE TITLE
Feature request: window position as hint

### ISSUE THREAD

**Issue by ltqsoft** [glfw/glfw#1603]
Should we provide window position before creating it?
`glfwWindowHint(GLFW_POSITION_X, 100);
glfwWindowHint(GLFW_POSITION_Y, 120);`

**Comment by ghuser404** [comment]
You can create a hidden window, position as you wish and then show it.

**Comment by elmindreda** [comment]
Agreed, and it may be a good thing to add to `glfwCreateWindow` in the next API revision.

**Comment by DerekCresswell** [comment]
I've got this implemented in a basic way on [my fork](https://github.com/DerekCresswell/glfw/tree/window-pos-hint).

This was just done with `window.c` so it should work will all platforms. This also allows for the default location of either X or Y to be applied if only one of the hints was given.  
There is the problem that for the first few frames (usually 4 main loops for me) the window would be in the default location. This creates an obvious jump.

I have been trying (and will continue) to get rid of this initial jump. It would appear to me that this requires setting up this implementation in each `_glfwPlatformCreateWindow` function.  I'm on Ubuntu and was experimenting but could not seem to get X11 to respect the values I passed in during `XWindowCreate`, perhaps because of the defaults being set later during `_glfwPlatformShowWindow`.

If the changes I made on my fork seem acceptable I'll turn it into a PR. Otherwise I would not mind any ideas or guidance on how to remove the jitter. I'll continue looking into this for a little while longer.

I'll also mention, I feel like adding this to `glfwCreateWindow` would be convenient, it might be getting to cluttered as a signature. Also forcing the user to chose a position could be awkward as  most of the time the window system chooses a specific position.

**Comment by flysand7** [comment]
The html docs on the website don't document this feature, perhaps it should be added there too?

**Comment by elmindreda** [comment]
@flysand7 [Done](https://www.glfw.org/docs/3.4/). Thanks!
